packages/sitecore-jss/src/layout/utils.ts

```diff
@@ -72,3 +79,41 @@ export function getChildPlaceholder(
 
   return rendering.placeholders[placeholderName];
 }
+
+/**
+ * Walks through layout rendering tree and invokes Reduce-compatible callback on every Rendering.
+ * Initial value determines return type
+ *
+ * @param {LayoutServiceData | RouteData<any> | ComponentRendering | HtmlElementRendering | null } object Layout, route data or rendering
+ * @param {Function} callback Reduce-style callback, accepting previous value and a rendering
+ * @param {any} initialValue Initial value that will be passed to callbacks, e.g. array
+ */
+export function reduceLayout<T>(
+  object: LayoutServiceData | RouteData<any> | ComponentRendering | HtmlElementRendering | null,
```

Comment:
Avoid using `any` (we went through a major cleanup of these a while back, and this _should_ fail linting)

Reply:
Well it's a different kind of any. It's a generic type parameter any, so it's not really that bad. But sure i'll expand that type